### DISCUSSION TITLE
HMIP-WRC2 - SHORT & LONG PRESS

### DISCUSSION THREAD

**Discussion by itn3rd77** [SukramJ/aiohomematic#85] [Q&A] [answered]
Hi Markus,

yesterday I switched completely to the new integration and I am very happy so far. Thanks for all your efforts! 

I just haven't figured out how I should handle the SHORT_PRESS and LONG_PRESS events e.g. on my HMIP-WRC2. The old integration had this special event-type `homematic.keypress` I used in the automations:

![grafik](https://user-images.githubusercontent.com/2655169/147205070-79496cbe-7bf9-4364-9734-eb8476cde8ba.png)

This is how the device looks in Home Assistant (virtual channels enabled):

![grafik](https://user-images.githubusercontent.com/2655169/147205681-6eb9db7c-5170-49dc-8c8c-6ccd99af1773.png)

And this one is from RaspberryMatic:

![grafik](https://user-images.githubusercontent.com/2655169/147205777-e271c779-f539-4fc9-8424-b600609c3a1c.png)

Thanks for your support.

Best regards
Ingo

**Comment by SukramJ** [comment, marked as answer]
Just select the device as trigger and then the trigger action.

**Reply by itn3rd77** [reply]
Thank you, exelent!

**Comment by eikowagenknecht** [comment]
For anyone looking for more details:

It is still handled with the homematic.keypress event. Just listen to that, then press the button and you see the exact events triggered ;-)

E.g.

hahomematic:
```
{
    "event_type": "homematic.keypress",
    "data": {
        "interface_id": "CCU3-HmIP-RF",
        "address": "001FDBE98A6CA2",
        "type": "press_long",
        "subtype": 1,
        "device_id": "589045dad0015bb76c01da3ac8b28478"
    },
    "origin": "LOCAL",
    "time_fired": "2022-01-12T12:13:27.028243+00:00",
    "context": {
        "id": "a7664f2170674351e9ac179fb4622da2",
        "parent_id": null,
        "user_id": null
    }
}
```

pyhomematic:
```
{
    "event_type": "homematic.keypress",
    "data": {
        "name": "EG.Flur Taster Bad (Gerät)",
        "param": "PRESS_LONG",
        "channel": 1
    },
    "origin": "LOCAL",
    "time_fired": "2022-01-12T12:13:27.037084+00:00",
    "context": {
        "id": "66fd1c5abf4032447e78c0e584156d2a",
        "parent_id": null,
        "user_id": null
    }
}
```

**Reply by ThomDietrich** [reply]
Hey all,
is there a reason why the `name` of the device is not provided by hahomematic? Given, the `address` is unique but quite ambiguous in yaml defined automation. I suppose there is no harm in providing the name alongside the address with every event!? Would be much appreciated.

CC @SukramJ

**Reply by SukramJ** [reply]
This integration uses `device_triggers` to make automations configurable via UI.
These values are used to match HA to HM 

```
        "interface_id": "CCU3-HmIP-RF",
        "address": "001FDBE98A6CA2",
        "type": "press_long",
        "subtype": 1,
```
The HM Name is not good for referencing.

**Reply by ThomDietrich** [reply]
Via UI that makes sense. In yaml rules this is quite error-prone to maintain:

```yaml
automation:
  - alias: Regel Badezimmer Lichtschalter lang
    trigger:
      - platform: event
        event_type: homematic.keypress
        event_data:
          address: 001FDBE98A6CA2
          type: press_long
    action:
```

Would it break your `device_triggers` if the additional field `name` was provided with the event? Being able to replace `address` by `name` would certainly be a huge improvement here.

**Reply by SukramJ** [reply]
Don't know. This must be checked.
Manual YAML automation feels like a huge step back.

**Reply by ThomDietrich** [reply]
Would appreciate if you could 👍

There is no best solution or step back. You have the dialog based way, many users prefer NodeRed, and many power users *need* the flexibility of yaml based configuration. Don't know about you but I maintain half a dozen instances with hundreds of automations. Not manageable via UI.

**Reply by ThomDietrich** [reply]
Was this ever resolved? Sorry, can't check atm
Thanks!
